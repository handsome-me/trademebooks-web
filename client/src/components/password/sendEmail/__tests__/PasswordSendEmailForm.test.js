import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import PasswordSendEmailForm from '../PasswordSendEmailForm'

describe('Component: PasswordSendEmailForm', () => {
  const props = {
    sendPasswordResetEmail: jest.fn()
  }

  const container = mount(<PasswordSendEmailForm {...props} />)

  it('should match the snapshot', () => {
    const passwordSendEmailFormSnapshot = renderer
      .create(<PasswordSendEmailForm {...props} />)
      .toJSON()
    expect(passwordSendEmailFormSnapshot).toMatchSnapshot()
  })

  it('should render the PasswordSendEmailForm component.', () => {
    const passwordSendEmailFormComponent = container.find(
      '.password-send-email-form'
    )
    expect(passwordSendEmailFormComponent.length).toBe(2)
  })

  it('should see the PasswordSendEmailForm component text like the header correctly.', () => {
    const passwordSendEmailFormComponent = container.find(
      '.password-send-email-form'
    )
    expect(passwordSendEmailFormComponent.find('.h2').first().text()).toBe(
      'Reset Your Password'
    )
  })

  it('should trigger a the PasswordSendEmailForm action/function when clicking on the send eamil button, in this case submitting the form.', () => {
    const passwordSendEmailFormComponent = container.find(
      '.password-send-email-form'
    )
    passwordSendEmailFormComponent.find('form').first().simulate('submit')
    expect(props.sendPasswordResetEmail).toHaveBeenCalled()
  })
})
