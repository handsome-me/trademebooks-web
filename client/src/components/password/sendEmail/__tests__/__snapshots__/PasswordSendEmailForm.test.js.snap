// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PasswordSendEmailForm should match the snapshot 1`] = `
<div
  className="container mt-4 password-send-email-form"
  data-test="container"
>
  <div
    className="row justify-content-center"
    data-test="row"
  >
    <div
      className="col-md-6"
      data-test="col"
    >
      <form
        onSubmit={[Function]}
      >
        <p
          className="h2 text-center mb-5"
        >
          Reset Your Password
        </p>
        <div
          className="grey-text"
        >
          <div
            className="md-form form-group"
          >
            <i
              className="fa fa-envelope prefix"
              data-test="fa"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            />
            <input
              aria-disabled={false}
              className="form-control"
              data-test="input"
              disabled={false}
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onInput={[Function]}
              required={true}
              type="email"
              value=""
            />
            <label
              aria-labelledby=""
              className=""
              data-error=""
              data-success=""
              id=""
              onClick={[Function]}
            >
              Type in your email
            </label>
          </div>
        </div>
        <div
          className="text-center"
        >
          <button
            className="btn-default btn Ripple-parent w-100"
            data-test="button"
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="submit"
          >
            Send Email
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
